src/bepasty/app.py

```diff
@@ -32,25 +33,63 @@ def __init__(self, app, prefix=''):
         self.app = app
         self.prefix = prefix
 
-    def __call__(self, environ, start_response):
+    def __call__(self, environ, start_response, set_script_name=True):
         if environ['PATH_INFO'].startswith(self.prefix):
             environ['PATH_INFO'] = environ['PATH_INFO'][len(self.prefix):]
-            environ['SCRIPT_NAME'] = self.prefix
+            if set_script_name:
+                environ['SCRIPT_NAME'] = self.prefix
             return self.app(environ, start_response)
         else:
             start_response('404', [('Content-Type', 'text/plain')])
             return ['This URL does not belong to the bepasty app.'.encode()]
 
 
+class ReverseProxied(PrefixMiddleware):
+    def __init__(self, app, scheme, server, script_name='', prefix=None):
+        self.app = app
+        self.scheme = scheme
+        self.server = server
+        self.script_name = script_name or ''
+        self.prefix = prefix
+        if prefix is not None:
+            super(ReverseProxied, self).__init__(app, prefix=prefix)
+
+    def __call__(self, environ, start_response):
+        scheme = environ.get('HTTP_X_FORWARDED_PROTO') or self.scheme
```

Comment:
rather use the default param of get:
```
environ.get(NAME, default)
```

Reply:
I already checked ProxyFix() while working on this. And main reason why I didn't use it is, I couldn't find way to integrate with PrefixMiddleware. And another reason is, changing unnecessary environment for this.

Reply:
Theoretically, middlewares should be stackable. Why did this not work?

Also, can you rephrase "And another reason is, changing unnecessary environment for this.", I did not understand what you mean.

Reply:
> Theoretically, middlewares should be stackable. Why did this not work?

both of PrefixMiddleware() and ProxyFix() modify SCRIPT_NAME, so those are conflict.

> Also, can you rephrase "And another reason is, changing unnecessary environment for this.", I did not understand what you mean.

ProxyFix() modifies REMOTE_ADDR, SERVER_NAME, and SERVER_PORT that we don't use those for now. and especially, to setup REMOTE_ADDR from X-Forwarded-For, we have to decide which part is trustable (hard coding or additional config are required).

Reply:
modifying SCRIPT_NAME by PrefixMiddleware is solved by your small change in there, right?

what I want to avoid is to invent yet another ProxyFixingMiddleware if werkzeug already has one.

that is code we need to maintain, it might contain unknown bugs. i'ld rather like to leave that to werkzeug, which likely has thousandfold more users than bepasty.

it is not a problem if ProxyFix does some stuff we do not use (yet).
also, iirc, it has a mechanism to only trust trustable headers.

Reply:
> modifying SCRIPT_NAME by PrefixMiddleware is solved by your small change in there, right?

hm, not sure. I will try again

> what I want to avoid is to invent yet another ProxyFixingMiddleware if werkzeug already has one.
> 
> that is code we need to maintain, it might contain unknown bugs. i'ld rather like to leave that to werkzeug, which likely has thousandfold more users than bepasty.

yes, then failed and copied to own one

> it is not a problem if ProxyFix does some stuff we do not use (yet).
> also, iirc, it has a mechanism to only trust trustable headers.

X-Forwarded-For is not trustable header. it is recording the path of proxies, and client can  behave as proxy to add fake X-Forwarded-For. So it is why ProxyFix() needs trustable number of ip in X-Forwarded-For.

Reply:
yes, that is known and werkzeug's `ProxyFix` has parameters to specify the number of trusted entries for all the headers it is processing.

i still do not see the reason why we need any other code than that. esp. considering i did not need any code to have bepasty running behind a proxy, not even ProxyFix.